### DISCUSSION TITLE
DSM, version: 7.2.2-72806

### DISCUSSION THREAD

**Discussion by EDub1492** [007revad/Synology_HDD_db#374] [General]
Answered in a previous thread

Thanks for the great work !!!

**Comment by 007revad** [comment]
Yes.

**Comment by TorbenSchreiter** [comment]
After completion of the DSM update to v7.2.2 the system initially booted up without having triggered the script scheduled for bootup. Therefore `/volume2` that has both all packages and the Synology Drive shared folders on them, were not there until I had manually run the script and rebooted the box a second time. 

Afterwards, everything seems to work, not however Synology Drive. It has reset itself to `/volume1` with virgin settings. Would it be safe to change in the UI the location as per the below or would this overwrite my Synology Drive DB already present - but not linked to - on `/volume2`? Or would I need to update symlinks via SSH?

![20241025 10_26_06-GLHV3 - Synology NAS](https://github.com/user-attachments/assets/7eb06934-c736-4d94-9795-ec4d35f67602)

EDIT: Actually, the above situation seems to have moved the entire package - not just the DB - back to `/volume1` from `/volume2`, where it was previously installed. Same for some other pkgs that definitely have been on `/volume2` before the DSM update such as Synology Photos or Advanced Media Extensions. Yet other pkgs however are still on `/volume2` (e.g. Cloud Sync & Hyper Backup). Strange.

![20241025 10_33_50-GLHV3 - Synology NAS](https://github.com/user-attachments/assets/2405efaf-5403-46b6-983c-d3ac5a2984b9)

Thanks for any advice!

**Comment by 007revad** [comment]
Changing the Location in Synology Drive Admin moves `@synologydrive` to the selected volume (overwriting any existing `@synologydrive` folder on that volume. I would do the following: 

1. Rename /volume2/@synologydrive to /volume2/@synologydrive_bak
2. Change the Location in Synology Drive Admin to volume 2.
3. Stop Synology Drive Server.
4. Rename the new /volume2/@synologydrive to /volume2/@synologydrive_new
5. Rename /volume2/@synologydrive_bak to /volume2/@synologydrive
6. Start Synology Drive Server.

**Reply by TorbenSchreiter** [reply]
Thanks, confirming this worked for me. What I did:

1. Stop Synology Drive Server
2. `sudo mv /volume2/@synologydrive /volume2/@synologydrive_bak`
3. Move the pkg using the latest release Synology_app_mover
4. Start Synology Drive Server
5. Change the Location in Synology Drive Admin UI to /volume 2
6. Stop Synology Drive Server
7. `sudo mv /volume2/@synologydrive /volume2/@synologydrive_new`
8. `sudo mv /volume2/@synologydrive_bak /volume2/@synologydrive`
9. Start Synology Drive Server

**Comment by TorbenSchreiter** [comment]
@007revad Post v.7.2.2 DSM update, this is the list of packages across volumes before re-moving any of them (as per the above). All of them were on `/volume2` prior to the DSM update.

Is it safe to move them again naively using Synology_app_mover or would this overwrite any pkg settings/DBs that might be on `/volume2` already?

![20241025 11_10_28-PuTTY](https://github.com/user-attachments/assets/26b9f0ce-9f28-4a29-ab62-1f56ae9990d8)

**Comment by 007revad** [comment]
Did the first 6 packages need repairing after the DSM update to 7.2.2?

**Reply by TorbenSchreiter** [reply]
From a users perspective, they were moved back to `/volume1` completely silently. No visible warning or anything. But now looking at the Log Center reveals pkg repairs (excerpt):
```
Info | System | 2024/10/25 10:15:53 | SYSTEM | [Synology Drive Server] service was started.
Info | System | 2024/10/25 10:15:05 | SYSTEM | [Cloud Sync] service was started.
Info | System | 2024/10/25 10:14:57 | SYSTEM | System successfully started [QuickConnect].
Info | System | 2024/10/25 10:14:31 | SYSTEM | System started to boot up.
Info | System | 2024/10/25 10:13:17 | SYSTEM | System successfully stopped [QuickConnect].
Info | System | 2024/10/25 10:12:45 | SYSTEM | [Synology Drive Server] service was stopped.
Info | System | 2024/10/25 10:12:39 | user123 | System started counting down to reboot.
Info | System | 2024/10/25 10:10:54 | SYSTEM | Package [Synology Photos] has been successfully repaired.
Info | System | 2024/10/25 10:10:54 | SYSTEM | System successfully started [Synology Photos].
Info | System | 2024/10/25 10:10:24 | SYSTEM | System successfully stopped [Synology Photos].
==> Info | System | 2024/10/25 10:09:52 | SYSTEM | Package [Synology Drive Server] has been successfully repaired.
Info | System | 2024/10/25 10:09:52 | SYSTEM | System successfully started [Synology Drive Server].
Info | System | 2024/10/25 10:09:48 | SYSTEM | [Synology Drive Server] service was started.
Info | System | 2024/10/25 10:09:36 | SYSTEM | System successfully stopped [Synology Drive Server].
==> Info | System | 2024/10/25 10:09:29 | SYSTEM | Package [Synology Application Service] has been successfully repaired.
==> Info | System | 2024/10/25 10:09:28 | SYSTEM | System successfully started [Synology Application Service].
==> Info | System | 2024/10/25 10:08:38 | SYSTEM | System successfully stopped [Synology Application Service].
Info | System | 2024/10/25 10:08:36 | SYSTEM | Package [Node.js v20] has been successfully repaired.
Info | System | 2024/10/25 10:08:36 | SYSTEM | System successfully started [Node.js v20].
Info | System | 2024/10/25 10:08:32 | SYSTEM | System successfully stopped [Node.js v20].
Info | System | 2024/10/25 10:08:28 | SYSTEM | Package [Universal Viewer] has been successfully repaired.
Info | System | 2024/10/25 10:08:28 | SYSTEM | System successfully started [Universal Viewer].
Info | System | 2024/10/25 10:08:25 | SYSTEM | System successfully stopped [Universal Viewer].
Info | System | 2024/10/25 10:08:21 | SYSTEM | Package [Advanced Media Extensions] has been successfully repaired.
Info | System | 2024/10/25 10:08:21 | SYSTEM | System successfully started [Advanced Media Extensions].
Info | System | 2024/10/25 10:07:44 | SYSTEM | System successfully stopped [Advanced Media Extensions].
Info | System | 2024/10/25 10:07:01 | SYSTEM | Package [Universal Search] has been successfully updated.
Info | System | 2024/10/25 10:06:50 | SYSTEM | Package [OAuth Service] has been successfully installed.
Info | System | 2024/10/25 10:06:48 | SYSTEM | Package [File Station] has been successfully repaired.
==> Warning | System | 2024/10/25 10:06:27 | SYSTEM | System detected unsupported [Cache device 2, Cache device 1].
Info | System | 2024/10/25 10:06:22 | SYSTEM | Package [SAN Manager] has been successfully updated.
Info | System | 2024/10/25 10:06:19 | SYSTEM | Package [Active Insight] has been successfully installed.
Info | System | 2024/10/25 10:06:18 | SYSTEM | Package [Python2] has been successfully updated.
Info | System | 2024/10/25 10:06:16 | SYSTEM | Package [Storage Manager] has been successfully repaired.
Info | System | 2024/10/25 10:06:10 | SYSTEM | System successfully started [QuickConnect].
Warning | System | 2024/10/25 10:06:09 | SYSTEM | System detected unsupported [Cache device 1].
Warning | System | 2024/10/25 10:06:09 | SYSTEM | System detected unsupported [Cache device 2].
==> Error | System | 2024/10/25 10:05:56 | SYSTEM | Storage Pool [2] has an insufficient number of drives.
Info | System | 2024/10/25 10:05:44 | SYSTEM | System started to boot up.
Info | System | 2024/10/25 10:03:26 | SYSTEM | System successfully stopped [QuickConnect].
Info | System | 2024/10/25 10:02:59 | SYSTEM | [Cloud Sync] service was stopped.
Info | System | 2024/10/25 10:02:56 | SYSTEM | [Synology Drive Server] service was stopped.
==> Info | System | 2024/10/25 10:02:50 | SYSTEM | Synology_HDD_db successfully updated to v3.5.103.
Info | System | 2024/10/25 10:02:47 | user123 | System started counting down to reboot.
==> Info | System | 2024/10/25 10:02:45 | user123 | Update was complete.
Info | System | 2024/10/25 10:00:39 | user123 | Package [Video Station] has been successfully uninstalled.
Info | System | 2024/10/25 09:55:48 | user123 | Download task for [DSM 7.2.2-72806] finished.
```

Surprisingly Synology_HDD_db was started 2024/10/25 10:02:50 _after_ update was complete. How come the M.2s still were not recognized?

**Reply by 007revad** [reply]
It looks like you have syno_hdd_db scheduled to run a shut down. You should schedule it to run at boot-up.

Those 6 packages would have been updated to DSM 7.2.2 versions, and the update "repaired" the packages due to the symlinks pointing to the missing volume 2.

It also updated storage manager from 1.0.0-00017 to 1.0.0-00502

**Reply by TorbenSchreiter** [reply]
There definitely are/were two tasks for Synology_HDD_db to be running on both bootup and shutdown. Could you maybe add a Log Center entry to be created by the script with a relevant msg that would explain whether all M.2 volumes should be present post script run vs. any script errors? E.g. "Synology_HDD_db completed (M.2 volume support [already] enabled [..])"?

Agree the missing volume probably triggered the pkg repairs. However, like said above: not sure why the volume was missing in the first place...

Storage Manager is now installed to "System Partition" according to the Package Center of v7.2.2. Not 100% sure if I remember correctly, but I actually think I had moved it also to `/volume2` last week as well. Maybe DSM v7.2.2 update wanted to move it from /volume2 to System Partition, did not find it due to missing volume and then repaired it?

Back to my question: how can I move the pkg 1-6 back to /volume2 without losing settings (except for Drive which I had already moved as per the above)?

**Reply by 007revad** [reply]
Synology Application Service, Node.js v20, Universal Viewer and Advanced Media Extensions will be fine be moved.

Synology Photos is the only one I'd be concerned about. If Synology Photos is currently showing your photos correctly then go ahead and move the Synology Photos package to volume 2.

**Reply by 007revad** [reply]
> Could you maybe add a Log Center entry

You can check the output of the last run of a scheduled user defined script in "Control Panel > Task Scheduler > select the task > Action > View Result > view details". Or set task scheduler to email you results each time it's run (which is what I have set).

**Reply by TorbenSchreiter** [reply]
Unfortunately, Synology Photos is currently not showing any photos any longer.

Confirming the move of the other 4x packages.

**Reply by TorbenSchreiter** [reply]
> > Could you maybe add a Log Center entry
> 
> You can check the output of the last run of a scheduled user defined script in "Control Panel > Task Scheduler > select the task > Action > View Result > view details". Or set task scheduler to email you results each time it's run (which is what I have set).


I had set up the emails, but had a faulty email server config, which is now fixed. Therefore nothing for the DSM update occasion, unfortunately.

Still think a log entry would make sense simply to see it at the right spot in the sequence of events in the log, but of course not a must.

**Reply by 007revad** [reply]
It would be handy to see the sequence of events. There would need to be a number of log entries. The downside is if someone's needs Synology Support's help the logs would show evidence they may not want support to see.

It's 10pm here. I'll get back to you tomorrow with steps to move Photos.

**Reply by 007revad** [reply]
What does this command return?
```
sudo readlink /var/services/photo
```

If you've moved the shared space folder to volume 2 try this: 
1. Go to Synology Photos > Settings > Shared Space.
2. Disable the Shared Space.
3. Then go back in and Enable Shared Space.

**Reply by TorbenSchreiter** [reply]
> What does this command return?
> 
> ```
> sudo readlink /var/services/photo
> ```

```
user123@GLHV3:~$ sudo readlink /var/services/photo
Password:
/volume1/photo
```
 
> If you've moved the shared space folder to volume 2 try this:
> 
> 1. Go to Synology Photos > Settings > Shared Space.
> 2. Disable the Shared Space.
> 3. Then go back in and Enable Shared Space.

I have not yet moved the pkg to `/volume2` yet, but `/photo` is on `/volume1` (pkg should go to `/volume2` though). Please confirm whether I should move the pkg first to `/volume2` before playing around w/ the Shared Space setting? This won't delete any contents of `/photo`, right?

**Reply by TorbenSchreiter** [reply]
Have moved the Synology Photo pkg to /volume2 now. Magically it now found the photos again. All good. Thanks.
